### PR TITLE
editor: improvements

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Removes editor service.
* Transfers the editor service functions to the editor components.
* Uses editor component to anothers components, extensions.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

### CODE REVIEW

**Review by vgranata — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 1]
```diff
@@ -118,13 +117,43 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
...
+
+  // Editor root field
+  public get rootField(): FormlyFieldConfig {
+    return this.rootFormlyConfig;
+  }
+
+  // Editor fonction
+  public get editorComponent(): () => EditorComponent {
+    return () => this;
+  }
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:149]: for my information : I don't really understand the return type of this function :(

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 2]
```diff
@@ -630,10 +653,113 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
...
+  /********** Form field manipulation **********/
+
+  /**
+   * Scroll to a field or fieldGroup and set focus in the first field candidate.
+   * @param field: the field or fieldGroup where to search about field candidate.
+   * @param scroll: is the screen should scroll to the field.
+   */
+    setFieldFocus(field: FormlyFieldConfig, scroll: boolean = false): boolean {
+    if (scroll === true && field.id)  {
+      const el = document.getElementById(`field-${field.id}`);
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:669]: indentation problem

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 3]
```diff
@@ -630,10 +653,113 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
...
+    }
+    field.focus = true;
+    return true;
+  }
+
+  /**
+   * Add a field to the hidden list
+   * @param field - FormlyFieldConfig, form config to be added
+   */
+   addHiddenField(field: FormlyFieldConfig): void {
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:691]: same

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 4]
```diff
@@ -630,10 +653,113 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
...
+      this._hiddenFields = this._hiddenFields.filter(f => f.id !== field.id);
+      this._hiddenFields.push(field);
+      this._hiddenFieldsSubject.next(this._hiddenFields);
+    }
+  }
+
+  /**
+   * Clear the list of the hidden fields
+   */
+   clearHiddenFields(): void {
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:704]: and again, and again, ... check next functions ;-)

**Review by zannkukai — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/editor/editor.component.ts:149]: Anonymous function that returns an instance of the EditorComponent

**Review by vgranata — APPROVED** [review]

**Comment by PascalRepond** [comment]
## Tests

- [ ] REP: patron editor, cannot edit user : JS error "ERROR TypeError: this.editorComponentInstance is undefined"

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**